### PR TITLE
Update README 

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR updates the README and:
- adds a high-level description of geomstats,
- distinguishes between pip3 and git clone installations of geomstats,
- adds a code snippet of an elementary example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
+This installs the latest version uploaded on PyPi.
+
+## Install geomstats via Git
+
+From a terminal (OS X & Linux), you can install geomstats and its requirements via Git as follows::
 
-To change backend:
 ```
-export GEOMSTATS_BACKEND=pytorch
+pip3 install -r requirements
```
Comment (nguigs) [inline README.md:33]: I think you have to clone first, otherwise you don't have the file requirements. And it should be requirements.txt

File: README.md [hunk 2]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
 
+## Install geomstats via pip3
 
-## Installation
-
-OS X & Linux:
+From a terminal (OS X & Linux), you can install geomstats and its requirements with ``pip3`` as follows::
 
 ```
 pip3 install -r requirements.txt
```
Comment (nguigs) [inline README.md:22]: First, you don't have the file requirements.txt on your machine to start with. Then I think running `pip3 install geomstats` alone installs all the dependencies and therefore is enough

File: README.md [hunk 3]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
+git clone https://github.com/geomstats/geomstats.git
 ```
 
-## Getting started
+This installs the latest GitHub version, useful for developers.
 
-Run example scripts, for example:
+## Choose the backend
+
+You can choose your backend by setting the environment variable ``GEOMSTATS_BACKEND`` to ``numpy``, ``tensorflow`` or ``pytorch``. By default, the numpy backend is used. You should only use the numpy backend for examples with visualizations.
```
Comment (nguigs) [inline README.md:41]: Not clear (implies that numpy only works for visualization).
--> 'The visualizations are only available with the numpy backend'

File: README.md [hunk 4]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
+To use ``geomstats`` for learning
+algorithms on Riemannian manifolds, you need to follow three steps:
+- instantiate the manifold of interest,
+- instantiate the learning algorithm of interest,
+- run the algorithm.
+
+The data should be represented by a ``gs.array``. This structure represents numpy arrays, or tensorflow/pytorch tensors, depending on the choice of backend.
+
+The following code snippet shows the use of tangent Principal Component Analysis on the
+space of 3D rotations, assuming ``data`` belongs to this space.
```
Comment (nguigs) [inline README.md:68]: data belong (plural)?

File: README.md [hunk 5]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
 ```
-nose2 tests
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+from geomstats.learning.pca import TangentPCA
+
+so3 = SpecialOrthogonal(n=3)
+metric = so3.bi_invariant_metric
+
+tpca = TangentPCA(metric=metric, n_components=2)
+tpca = tpca.fit(data, base_point=metric.mean(data))
```
Comment (nguigs) [inline README.md:78]: data should be defined first:
`data = so3.random_uniform(10)`
Also, is this code old? `metric.mean(data)` won't work, just pass `tpca.fit(data)`

File: README.md [hunk 6]
```diff
@@ -11,44 +14,78 @@ Computations and statistics on manifolds with geometric structures.
...
+- instantiate the manifold of interest,
+- instantiate the learning algorithm of interest,
+- run the algorithm.
+
+The data should be represented by a ``gs.array``. This structure represents numpy arrays, or tensorflow/pytorch tensors, depending on the choice of backend.
+
+The following code snippet shows the use of tangent Principal Component Analysis on the
+space of 3D rotations, assuming ``data`` belongs to this space.
+
 ```
```
Comment (nguigs) [inline README.md:70]:
```python ?

File: README.md (hunk 5 above)
Comment (ninamiolane) [reply README.md]: Changed. Thanks!

**Comment by ninamiolane** [comment]
Ok, let me add an issue for the optional pytorch and tensorflow requirements, so we can deal with is separately. But this seems to be helpful:
https://stackoverflow.com/questions/3664478/optional-dependencies-in-a-pip-requirements-file?lq=1
